src/components/form/form.scss

```diff
@@ -10,12 +10,112 @@
     }
 }
 
+.limel-form-layout--default {
+    --gap: 1rem;
+    display: grid;
+    gap: var(--gap);
+}
+
 .limel-form-layout--grid {
+    --gap: 1rem;
+    --min-height-of-one-row: 4.71rem;
     display: grid;
-    grid-template-columns: repeat(auto-fill, minmax(18.75rem, 1fr));
-    gap: 0.5rem;
+    padding: var(--gap);
+    gap: var(--gap);
+    grid-template-columns: repeat(var(--number-of-columns), 1fr);
+
+    &.auto-reorder-to-avoid-empty-cells {
+        grid-auto-flow: dense;
+    }
+
+    &[style*='--number-of-columns:5;'],
+    &.layout-5-columns {
+        grid-column-gap: calc(var(--gap) + 1rem);
+    }
+    &[style*='--number-of-columns:4;'],
+    &.layout-4-columns {
+        grid-column-gap: calc(var(--gap) + 0.75rem);
+    }
+    &[style*='--number-of-columns:3;'],
+    &.layout-3-columns {
+        grid-column-gap: calc(var(--gap) + 0.5rem);
+    }
+    &[style*='--number-of-columns:2;'],
+    &.layout-2-columns {
+        grid-column-gap: calc(var(--gap) + 0.25rem);
+    }
```

Comment:
Why is there a bigger gap when there are more columns? 🤔

Is it to make the columns easier to distinguish? I guess maybe my real question is: "Why is there a _smaller_ gap when there are fewer columns?" 😄

Reply:
Good question and it tickles, because I try to be smart here. But I'm not sure if I am 😄 

My assumption is:
When users choose a 5 columned layout, it means that the UI has enough space to fit-in 5 columns. Therefore, the UI can as well be more "airy" (to improve readability of the content and provide more space for the eyes to rest).

(Same thing when they don't deliberately specify number of columns, but the form decides that it can display 5 columns because there is enough space. This will essentially make the UI more compact on smaller containers or devices.)

What do you think?

Reply:
Rule of thumb: If it feels clever, it's a red flag.

Reducing complexity, keeping things simple, is the most important thing we can do. We need to be minimalist not only in our visual design, but in terms of code complexity as well. Let's go with the smaller gap for now.

Reply:
> Let's go with the smaller gap for now.

If this means you wanna delete those lines, it's fine for me. ;)

Reply:
Yes, sorry, wasn't very clear there. I've indeed removed them. I've just forgotten to push 😄